src/main/java/org/jabref/gui/entryeditor/AiChatTab.java

```diff
@@ -0,0 +1,238 @@
+package org.jabref.gui.entryeditor;
+
+import java.util.List;
+
+import javafx.geometry.Insets;
+import javafx.geometry.NodeOrientation;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.control.Tooltip;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+
+import org.jabref.gui.DialogService;
+import org.jabref.logic.ai.AiChat;
+import org.jabref.logic.ai.AiChatData;
+import org.jabref.logic.ai.AiConnection;
+import org.jabref.logic.ai.AiIngestor;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.preferences.AiPreferences;
+import org.jabref.preferences.FilePreferences;
+import org.jabref.preferences.PreferencesService;
+
+import com.tobiasdiez.easybind.EasyBind;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.UserMessage;
+
+public class AiChatTab extends EntryEditorTab {
+    public static final String NAME = "AI chat";
+
+    private final DialogService dialogService;
+
+    private final FilePreferences filePreferences;
+    private final AiPreferences aiPreferences;
+
+    private final BibDatabaseContext bibDatabaseContext;
+
+    private AiConnection aiConnection = null;
+    private AiChat aiChat = null;
+
```

Comment:
This applies for every-every field?
What if I have these fields in a class:
- Logger
- Data fields
- Constants

I can't group them with new lines? And what will be the right order for this fields?

Reply:
checkstyle will tell about the order. Grouping is OK, add comments then. However, this could make the code more unreadable. Just keep one empty line between each "thing".

If reasonable, you could create a nested `record` class for more clearer grouping. However, in your case, this doesn't seem to be necessary.